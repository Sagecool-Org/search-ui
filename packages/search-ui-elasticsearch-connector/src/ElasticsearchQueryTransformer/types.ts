import type {
  QueryDslQueryContainer,
  SearchRequest,
  SearchHighlight
} from "@elastic/elasticsearch/lib/api/types";

export type Filter = Pick<QueryDslQueryContainer, "bool">;
export type Sort = SearchRequest["sort"];

export type FilterValue = Pick<QueryDslQueryContainer, "term">;
export type FilterValueRange = Pick<QueryDslQueryContainer, "range">;

export type RequestBody = SearchRequest;

export type Query = Pick<QueryDslQueryContainer, "bool">;

export type HighlightFields = Pick<SearchHighlight, "fields">;
export type Aggregations = Record<string, any>;
