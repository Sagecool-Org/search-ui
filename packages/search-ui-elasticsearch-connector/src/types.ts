import type { estypes } from "@elastic/elasticsearch";
import { QueryConfig, RequestState } from "@elastic/search-ui";

export type SearchRequest = estypes.SearchRequest;
export type SearchSourceFilter = estypes.SearchSourceFilter;
export type SearchHit = estypes.SearchHit;

export type PostProcessRequestBodyFn = (
  requestBody: SearchRequest,
  requestState: RequestState,
  queryConfig: QueryConfig
) => SearchRequest;

export interface CloudHost {
  id: string;
}

export type ConnectionOptions = {
  host?: string;
  cloud?: CloudHost;
  index: string;
  apiKey?: string;
  connectionOptions?: {
    headers?: Record<string, string>;
  };
};
