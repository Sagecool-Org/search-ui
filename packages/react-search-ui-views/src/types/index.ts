import type { FacetValue, FieldValue, FilterType } from "@elastic/search-ui";

export interface BaseContainerProps {
  children?: React.ReactNode;
  className?: string;
}

export type { ErrorBoundaryViewProps } from "../ErrorBoundary";

export type { SearchBoxAutocompleteViewProps } from "../Autocomplete";

export type {
  PagingViewProps,
  PagingContainerContext,
  PagingContainerProps
} from "../Paging";

export type {
  PagingInfoContainerProps,
  PagingInfoContainerContext,
  PagingInfoViewProps
} from "../PagingInfo";

export type { ResultsContainerProps, ResultsViewProps } from "../Results";

export type { ResultContainerProps, ResultViewProps } from "../Result";

export type {
  ResultsPerPageContainerContext,
  ResultsPerPageContainerProps,
  ResultsPerPageViewProps
} from "../ResultsPerPage";

export type {
  SearchBoxContainerContext,
  SearchBoxContainerProps,
  SearchBoxViewProps
} from "../SearchBox";

export type { InputViewProps } from "../SearchInput";

export type { SortingContainerProps, SortingViewProps } from "../Sorting";

export type FacetViewProps = {
  className?: string;
  label: string;
  onMoreClick: () => void;
  onRemove: (value: FieldValue) => void;
  onChange: (value: FieldValue) => void;
  onSelect: (value: FieldValue) => void;
  options: FacetValue[];
  showMore: boolean;
  values: any;
  showSearch: boolean;
  onSearch: (value: string) => void;
  searchPlaceholder: string;
};

export type FacetContainerProps = BaseContainerProps & {
  filterType?: FilterType;
  show?: number;
  view?: React.ComponentType<FacetViewProps>;
  isFilterable?: boolean;
  field: string;
  label: string;
  persistent?: boolean;
};

// From SO https://stackoverflow.com/a/59071783
// TS Utility to rename keys in a type
// Usage type x = Rename<{ renameMe: string }, { "renameMe": "newName" }>
// x === { newName: string }
export type Rename<
  T,
  R extends {
    [K in keyof R]: K extends keyof T ? PropertyKey : "Error: key not in T";
  }
> = { [P in keyof T as P extends keyof R ? R[P] : P]: T[P] };
