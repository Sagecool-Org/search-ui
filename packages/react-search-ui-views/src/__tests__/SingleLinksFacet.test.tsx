import React from "react";
import { render } from "@testing-library/react";
import SingleLinksFacet from "../SingleLinksFacet";
import { FacetViewProps } from "../types";

const params: FacetViewProps = {
  label: "Facet",
  onRemove: jest.fn(),
  onSelect: jest.fn(),
  options: [
    { value: "1", count: 1, selected: false },
    { value: "2", count: 1, selected: false }
  ],
  onChange: jest.fn(),
  onMoreClick: jest.fn(),
  onSearch: jest.fn(),
  searchPlaceholder: "Search",
  showMore: false,
  showSearch: false,
  values: []
};

it("renders correctly", () => {
  const { container } = render(<SingleLinksFacet {...params} />);
  expect(container).toMatchSnapshot();
});

it("renders falsey values correctly", () => {
  const { container } = render(
    <SingleLinksFacet
      {...params}
      options={[
        {
          value: 0,
          count: 10,
          selected: false
        },
        {
          value: false,
          count: 20,
          selected: false
        },
        {
          value: "",
          count: 30,
          selected: false
        }
      ]}
    />
  );
  expect(container).toMatchSnapshot();
});

describe("determining selected option", () => {
  it("will correctly determine which of the options is selected", () => {
    const { container } = render(
      <SingleLinksFacet
        {...params}
        options={[
          { value: "1", count: 1, selected: true },
          { value: "2", count: 1, selected: false }
        ]}
      />
    );
    const listItems = container.querySelectorAll("li");
    expect(listItems).toHaveLength(1);
    expect(listItems[0]).toHaveTextContent("1 (Remove)");
  });

  it("will used the first selected option when multiple options are selected", () => {
    const { container } = render(
      <SingleLinksFacet
        {...params}
        options={[
          { value: "1", count: 1, selected: true },
          { value: "2", count: 1, selected: true }
        ]}
      />
    );
    const listItems = container.querySelectorAll("li");
    expect(listItems).toHaveLength(1);
    expect(listItems[0]).toHaveTextContent("1 (Remove)");
  });

  it("will correctly determine when no value is selected", () => {
    const { container } = render(
      <SingleLinksFacet
        {...params}
        options={[
          { value: "1", count: 1, selected: false },
          { value: "2", count: 1, selected: false }
        ]}
      />
    );
    expect(container.querySelectorAll("li")).toHaveLength(2);
  });
});

it("renders with className prop applied", () => {
  const customClassName = "test-class";
  const { container } = render(
    <SingleLinksFacet {...params} className={customClassName} />
  );
  expect(container.firstChild).toHaveClass("sui-facet", "test-class");
});
