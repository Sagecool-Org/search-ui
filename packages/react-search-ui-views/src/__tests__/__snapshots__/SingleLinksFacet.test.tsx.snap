// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sui-facet"
  >
    <div>
      <div
        class="sui-facet__title"
      >
        Facet
      </div>
      <ul
        class="sui-single-option-facet"
      >
        <li
          class="sui-single-option-facet__item"
        >
          <a
            class="sui-single-option-facet__link"
            data-transaction-name="facet - Facet"
            href="/"
          >
            1
          </a>
           
          <span
            class="sui-facet__count"
          >
            1
          </span>
        </li>
        <li
          class="sui-single-option-facet__item"
        >
          <a
            class="sui-single-option-facet__link"
            data-transaction-name="facet - Facet"
            href="/"
          >
            2
          </a>
           
          <span
            class="sui-facet__count"
          >
            1
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`renders falsey values correctly 1`] = `
<div>
  <div
    class="sui-facet"
  >
    <div>
      <div
        class="sui-facet__title"
      >
        Facet
      </div>
      <ul
        class="sui-single-option-facet"
      >
        <li
          class="sui-single-option-facet__item"
        >
          <a
            class="sui-single-option-facet__link"
            data-transaction-name="facet - Facet"
            href="/"
          >
            0
          </a>
           
          <span
            class="sui-facet__count"
          >
            10
          </span>
        </li>
        <li
          class="sui-single-option-facet__item"
        >
          <a
            class="sui-single-option-facet__link"
            data-transaction-name="facet - Facet"
            href="/"
          >
            false
          </a>
           
          <span
            class="sui-facet__count"
          >
            20
          </span>
        </li>
        <li
          class="sui-single-option-facet__item"
        >
          <a
            class="sui-single-option-facet__link"
            data-transaction-name="facet - Facet"
            href="/"
          />
           
          <span
            class="sui-facet__count"
          >
            30
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
